src/main/java/com/microsoft/aad/msal4j/Authority.java

```diff
@@ -76,35 +78,56 @@ static AuthorityType detectAuthorityType(URL authorityUrl) {
         final String firstPath = path.substring(0, path.indexOf("/"));
 
 
-        if(isB2CAuthority(firstPath)){
+        if (isB2CAuthority(firstPath)) {
             return AuthorityType.B2C;
-        } else if(isAdfsAuthority(firstPath)) {
+        } else if (isAdfsAuthority(firstPath)) {
             return AuthorityType.ADFS;
         } else {
             return AuthorityType.AAD;
         }
     }
 
-    void validateAuthorityUrl() {
-        if (!this.canonicalAuthorityUrl.getProtocol().equalsIgnoreCase("https")) {
+    static void validateAuthority(URL authorityUrl) {
+        if (!authorityUrl.getProtocol().equalsIgnoreCase("https")) {
             throw new IllegalArgumentException(
                     "authority should use the 'https' scheme");
```

Comment:
Might be a good opportunity to move these error messages to AuthenticationErrorMessage or a constants file.

Reply:
it is not AuthenticationErrorMessage it is IllegalArgumentException message - going to add corresponding constants